flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -624,15 +626,24 @@ private void setupResultPartitionBookkeeping(
 			JobID jobId,
 			Collection<ResultPartitionDeploymentDescriptor> producedResultPartitions,
 			CompletableFuture<ExecutionState> terminationFuture) {
-		final List<ResultPartitionID> partitionsRequiringRelease = filterPartitionsRequiringRelease(producedResultPartitions);
+		final Set<ResultPartitionDeploymentDescriptor> partitionsRequiringRelease = filterPartitionsRequiringRelease(producedResultPartitions);
```

Comment:
in this commit, this is not quite clear why we switch from `ResultPartitionID` to `ResultPartitionDeploymentDescriptor`

Reply:
One could move this change into a separate commit.